### PR TITLE
[feature] merge mysql operator code to the radondb main branch

### LINKED ISSUE #117: [feature] add operator sidecar
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

To process some operators as a init container.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by zhyass**
### What type of PR is this?

/enhancement

### Which issue(s) this PR fixes?

Fixes #117 #118 #119 #125 #128 

### What this PR does?

Summary:
Update the cluster api, add sidecar container, add status api, update the config files, add helm files, update the Dockerfile, Makefile...

### Special notes for your reviewer?

For test, run:
```
helm install test charts/mysql-operator/
kubectl apply -f config/samples/mysql_v1alpha1_cluster.yaml
```

Uninstall:
```
kubectl delete -f config/samples/mysql_v1alpha1_cluster.yaml
helm uninstall test
kubectl delete customresourcedefinitions.apiextensions.k8s.io clusters.mysql.radondb.com
```

### CODE REVIEW

**Comment by andyli029** [comment]
Ping @aceking @hustjieke

**Comment by hustjieke** [comment]
ACK.

**Comment by acekingke** [comment]
ACK

**Review by acekingke — COMMENTED** [review]
ok

File: sidecar/init.go [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+func checkIfPathExists(path string) (bool, error) {
+	_, err := os.Open(path)
+
+	if os.IsNotExist(err) {
+		return false, nil
+	} else if err != nil {
+		log.Error(err, "failed to open file", "file", path)
+		return false, err
+	}
+
```
[the remark this answers is not in the document]
Comment (zhyass) [reply sidecar/init.go:153]: good suggest

**Pull request merged**